contracts/Remittance.sol

```diff
@@ -1,16 +1,118 @@
 //SPDX-License-Identifier: MIT
 pragma solidity >= 0.6.0 < 0.7.0;
 
-import "../node_modules/@openzeppelin/contracts/access/Ownable.sol";
-import "../node_modules/@openzeppelin/contracts/utils/Pausable.sol";
+import "./access/Ownable.sol";
+import "./utils/Pausable.sol";
 
-import {SafeMath} from "../node_modules/@openzeppelin/contracts/math/SafeMath.sol";
+import {SafeMath} from "./math/SafeMath.sol";
 
 contract Remittance is Ownable, Pausable {
 
-    using SafeMath for uint;
+    using SafeMath for uint256;
 
+    mapping(address => uint) public balances;
 
+    mapping(bytes32 => RemittanceInstance) public remittances;
+
+    struct RemittanceInstance {
+        address funder;
+        address broker;
+        uint amount;
+        bool fundsReleased;
+    }
+
+    event Deposit(
+        address indexed depositor,
+        uint amount
+    );
+
+    event RemittanceCreated(
+        bytes32 indexed hash,
+        address indexed funder,
+        address indexed broker,
+        uint amount
+    );
+
+    event RemittanceFundsReleased(
+        bytes32 indexed hash,
+        address indexed broker,
+        uint amount
+    );
+
+    event WithDraw(
+        address indexed withdrawer,
+        uint amount
+    );
+
+    // Generic deposit function
+    function deposit() external payable whenNotPaused {
+        require(msg.value > 0, "The value must be greater than 0");
+        balances[msg.sender] = balances[msg.sender].add(msg.value);
+
+        emit Deposit(msg.sender, msg.value);
+    }
+
+    // The funder can create a remittance
+    function create(bytes32 hash, address broker, uint256 amount)
+        external
+        whenNotPaused
+        returns(bool success)
+    {
+        require(broker != msg.sender, "The caller cannot be the broker");
+        require(amount > 0, "The amount must be greater than 0");
+
+        uint256 funderBalance = balances[msg.sender];
+        require(funderBalance >= amount, "There are insufficient funds in the funder's account to create this remittance");
+
+        // Subtract from Funder balance
+        balances[msg.sender] = SafeMath.sub(funderBalance, amount);
+
+        RemittanceInstance storage remittanceInstance = remittances[hash];
+
+        remittanceInstance.funder = msg.sender;
```

Comment:
I see `.funder` be set, but is it ever used again?

Reply:
> I see `.funder` be set, but is it ever used again?

It is not used again. My intention is that there would be a record on the blockchain that funder X had sent a remittance to broker Y. Is the funder address (msg.sender) information explicit somewhere else on the blockchain in relation to this remittance? For example, the address calling the 'create' function is stored?

Reply:
As for keeping this information, it is saved in the creation event, and that's probably good enough, depending on your use case.

Reply:
Sorry. I mean to ask about this in our Zoom session.
 
 `it is saved in the creation event`
 
 Is this a reference to the logged emitted event? I understood this to me non-permanent. Will this eventually be lost?

Reply:
Yes, in the event.
Not lost if the blockchain keeps everything. But at the rate it is going, ways will be found to reduce its size. The first idea is to have storage rent such that you would need to pay for the storage to be kept in the future. So your `.funder` would also be at risk of being erased.
As for events being lost, it is also possible. There is talk of roll-ups in other areas, and transaction receipts could also "fall prey" to them, with consequences I cannot fully grasp yet.